manager/allocator/networkallocator/networkallocator.go

```diff
@@ -303,38 +324,84 @@ func OnInit(options *ServiceAllocationOpts) {
 // IsServiceAllocated returns if the passed service has its network resources allocated or not.
 // init bool indicates if the func is called during allocator initialization stage.
 func (na *NetworkAllocator) IsServiceAllocated(s *api.Service, flags ...func(*ServiceAllocationOpts)) bool {
-	var options ServiceAllocationOpts
+	return !na.IsServiceToBeAllocated(s, flags)
+}
 
+// IsServiceToBeAllocated returns true if the passed service needs to have network resources allocated/updated.
+func (na *NetworkAllocator) IsServiceToBeAllocated(s *api.Service, flags []func(*ServiceAllocationOpts)) bool {
+	var options ServiceAllocationOpts
 	for _, flag := range flags {
 		flag(&options)
 	}
 
+	// Always prefer NetworkAttachmentConfig in the TaskSpec
+	specNetworks := s.Spec.Task.Networks
+	if len(specNetworks) == 0 && s != nil && len(s.Spec.Networks) != 0 {
```

Comment:
if `s` was nil, previous line would have paniced.
Is it possible it is nil when this code is called ?